spring-cloud-contract-wiremock/src/main/java/org/springframework/cloud/contract/wiremock/WireMockTestExecutionListener.java

```diff
@@ -93,7 +97,12 @@ public void afterTestMethod(TestContext testContext) throws Exception {
 	}
 
 	private boolean annotationMissing(TestContext testContext) {
-		if (AnnotationUtils.findAnnotation(testContext.getTestClass(), AutoConfigureWireMock.class) == null) {
+		Annotation annotation = MergedAnnotations
+				.from(testContext.getTestClass(), MergedAnnotations.SearchStrategy.TYPE_HIERARCHY_AND_ENCLOSING_CLASSES,
```

Comment:
We should consider submitting a PR to Spring `AnnotationUtils.findAnnotation` to allow specifying a search strategy. 

There are no other usages of `AnnotationUtils.findAnnotation` in SCC or I would have abstracted this out into SCC AnnotationUtils until it was available in Spring AU.

Reply:
cc @sbrannen ?

Reply:
The proper (and only supported) way to find annotations on a test class, in its type hierarchy, or in its enclosing class hierarchy is to use `org.springframework.test.context.TestContextAnnotationUtils` which provides utilities that transparently honor `@NestedTestConfiguration` semantics.

The `MergedAnnotations.SearchStrategy.TYPE_HIERARCHY_AND_ENCLOSING_CLASSES` strategy is effectively broken and unsupported.

Furthermore, we have no plans to modify `AnnotationUtils` "to allow specifying a search strategy".

Reply:
Thx @sbrannen - much cleaner now,

> is to use org.springframework.test.context.TestContextAnnotationUtils which provides utilities that transparently honor @NestedTestConfiguration semantics

Nice! I looked for `TestAnnotationUtils*` in IDEA but did not think to look for `TestContext**` variant.  

New tool in my toolbox:
```
TestContextAnnotationUtils is to AnnotationUtils as ReflectionTestUtils is to ReflectionUtils
```

>The MergedAnnotations.SearchStrategy.TYPE_HIERARCHY_AND_ENCLOSING_CLASSES strategy is effectively broken and unsupported.

Good to know. Are there plans to `@deprecate` the enum value? 

@marcingrzejszczak I have updated accordingly.

Reply:
> Thx @sbrannen - much cleaner now.

Glad I could help.

> Nice! I looked for `TestAnnotationUtils*` in IDEA but did not think to look for `TestContext**` variant.

Yeah, I had unfortunately already used that name in `org.springframework.test.annotation.TestAnnotationUtils` and didn't want to reuse the name, so I went for `TestContextAnnotationUtils` for the new support.

> ```
> TestContextAnnotationUtils is to AnnotationUtils as ReflectionTestUtils is to ReflectionUtils
> ```

Sure, that's one way to look at it, though `ReflectionTestUtils` is intended to be used by developers in their own tests; whereas, `TestContextAnnotationUtils` is primarily only suited for framework code.

> > The MergedAnnotations.SearchStrategy.TYPE_HIERARCHY_AND_ENCLOSING_CLASSES strategy is effectively broken and unsupported.
> 
> Good to know. Are there plans to `@deprecate` the enum value?

We actually considered doing that for Spring Framework 5.3 but didn't go through with it since some people might actually be using it for special use cases. It's just that the resulting search algorithm does not provide useful semantics within the core Spring Framework itself. That's why I had to introduce `TestContextAnnotationUtils` in order to provide meaningful semantics within `spring-test`.

Reply:
Thx again for the background info and details.